osu.Server.Queues.ScoreStatisticsProcessor/Processors/RankedScoreProcessor.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Diagnostics;
+using JetBrains.Annotations;
+using MySqlConnector;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring.Legacy;
+using osu.Server.Queues.ScoreStatisticsProcessor.Helpers;
+using osu.Server.Queues.ScoreStatisticsProcessor.Models;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Processors
+{
+    [UsedImplicitly]
+    public class RankedScoreProcessor : IProcessor
+    {
+        public bool RunOnFailedScores => false;
+
+        public void RevertFromUserStats(SoloScoreInfo score, UserStats userStats, int previousVersion, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            if (!score.Passed)
+                return;
+
+            if (!DatabaseHelper.IsBeatmapValidForRankedCounts(score.BeatmapID, conn, transaction))
```

Comment:
One interesting thing here is that with these checks, we can't use revert for the case of beatmap unrank / deletion. I'm not sure we'd want to, but felt it worth mentioning.

Reply:
Yeah I'm not sure. I'm not very well acquainted with those processes so I don't feel like I can suggest anything at the time, but if you think it important I can spend some time finding out how that works so that I can propose alternatives.